### PR TITLE
Add msstore productId to export manifest and to wingetutil interop manifest object

### PR DESCRIPTION
**Pull request by yao-msft**
This change was needed as part of the effort to make exported manifest (from winget download msstore apps) to be re-parsed and uploaded to winget rest source.
 
Updated the some of the manifest schemas to add "required" for some required fields previously missing.

Also, during my cross validation with the manifest schemas, I found ExpectedReturnCode::Custom was added in manifest but not actually updated in code. So I updated the ExpectedReturnCode map in code with this change too.

### CODE REVIEW

File: src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw [hunk 1]
```diff
@@ -1284,6 +1284,9 @@ Do you agree to the terms?</value>
   <data name="InstallFlowReturnCodeSystemNotSupported" xml:space="preserve">
     <value>The current system configuration does not support the installation of this package.</value>
   </data>
+  <data name="InstallFlowReturnCodeCustomError" xml:space="preserve">
+    <value>Installation failed with installer custom error. Contact package support.</value>
```
Comment (ryfu-msft) [inline src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:1288]: nit:
```suggestion
    <value>Installation failed with a custom installer error. Contact package support.</value>

```

File: src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw [hunk 2]
```diff
@@ -2521,6 +2524,9 @@ Please specify one of them using the --source option to proceed.</value>
   <data name="APPINSTALLER_CLI_ERROR_INSTALL_UPGRADE_NOT_SUPPORTED" xml:space="preserve">
     <value>The installer does not support upgrading an existing package.</value>
   </data>
+  <data name="APPINSTALLER_CLI_ERROR_INSTALL_CUSTOM_ERROR" xml:space="preserve">
+    <value>Installation failed with installer custom error.</value>
```
Comment (ryfu-msft) [inline src/AppInstallerCLIPackage/Shared/Strings/en-us/winget.resw:2528]:
```suggestion
    <value>Installation failed with a custom installer error.</value>

```

File: src/AppInstallerSharedLib/Errors.cpp [hunk 3]
```diff
@@ -244,6 +244,7 @@ namespace AppInstaller
             WINGET_HRESULT_INFO(APPINSTALLER_CLI_ERROR_INSTALL_INVALID_PARAMETER, "Invalid parameter."),
             WINGET_HRESULT_INFO(APPINSTALLER_CLI_ERROR_INSTALL_SYSTEM_NOT_SUPPORTED, "Package not supported by the system."),
             WINGET_HRESULT_INFO(APPINSTALLER_CLI_ERROR_INSTALL_UPGRADE_NOT_SUPPORTED, "The installer does not support upgrading an existing package."),
+            WINGET_HRESULT_INFO(APPINSTALLER_CLI_ERROR_INSTALL_CUSTOM_ERROR, "Installation failed with installer custom error."),
```
Comment (ryfu-msft) [inline src/AppInstallerSharedLib/Errors.cpp:247]:
```suggestion
            WINGET_HRESULT_INFO(APPINSTALLER_CLI_ERROR_INSTALL_CUSTOM_ERROR, "Installation failed with a custom installer error."),

```

**Review by ryfu-msft — DISMISSED** [review]

**Review by ryfu-msft — APPROVED** [review]

**Comment by yao-msft** [comment]
@microsoft-github-policy-service agree

**Pull request merged**